contracts/collections/sg721-base/src/contract.rs

```diff
@@ -299,12 +299,15 @@ where
 
         self.parent.increment_tokens(deps.storage)?;
 
-        Ok(Response::new()
+        let mut res = Response::new()
             .add_attribute("action", "mint")
             .add_attribute("minter", info.sender)
             .add_attribute("owner", msg.owner)
-            .add_attribute("token_id", msg.token_id)
-            .add_attribute("token_uri", msg.token_uri.unwrap_or_default()))
+            .add_attribute("token_id", msg.token_id);
+        if msg.token_uri.is_some() {
+            res = res.add_attribute("token_uri", msg.token_uri.unwrap_or_default());
```

Comment:
i think the indexer is expecting a `token_uri` field. was this added to deal with default `token_uri`?

Reply:
This is one thing we wanted to discuss because ->

The sg721-metadata-on-chain inherits from sg721_base: https://github.com/public-awesome/launchpad/blob/33270c8c89451f680fe19476ab64041a9eb0099b/contracts/collections/sg721-metadata-onchain/src/lib.rs#L10

And given an on-chain metadata does not have a token_uri, it crashes at the mint at this specific place. 

We can modify sg721-metadata-onchain `mint` if needed, we just need to agree on how it should be designed.